hconfig/mapl3hconfig_get_private.F90

```diff
@@ -0,0 +1,107 @@
+#include "MAPL_ErrLog.h"
+module mapl3hconfig_get_private
+   use :: esmf, only: ESMF_HConfig, ESMF_HConfigIsDefined, MAXSTRLEN => ESMF_MAXSTR
+   use :: esmf, only: ESMF_HConfigAsI4, ESMF_KIND_I4, ESMF_HConfigAsI8, ESMF_KIND_I8
+   use :: esmf, only: ESMF_HConfigAsR4, ESMF_KIND_R4, ESMF_HConfigAsR8, ESMF_KIND_R8
+   use :: esmf, only: ESMF_HConfigAsLogical, ESMF_HConfigAsString
+   use mapl_ErrorHandling
+   use mapl_KeywordEnforcer
+
+   implicit none
+
+   public :: MAXSTRLEN
+   public :: get_value
+
+   interface get_value
+      module procedure :: get_value_scalar
+   end interface get_value
+
+contains
+
+   subroutine get_value_scalar(hconfig, value, found, message, keystring, rc)
+      type(ESMF_HConfig), intent(inout) :: hconfig
+      class(*), intent(inout) :: value
+      logical, intent(out) :: found
+      character(len=:), allocatable, intent(inout) :: message
+      character(len=*), intent(in) :: keystring
+      integer, intent(out) :: rc
+
+      character(len=:), allocatable :: typestring
+      character(len=:), allocatable :: valuestring
+
+      integer :: status
+      integer :: ios
+      character(len=MAXSTRLEN) :: rawstring
+
+      found = ESMF_HConfigIsDefined(hconfig, keystring=keystring, _RC)
+      if(.not. found) then
+         _RETURN(_SUCCESS)
+      end if
+
+      select type(value)
+      type is (integer(kind=ESMF_KIND_I4))
+         typestring = 'I4'
+         value = ESMF_HConfigAsI4(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(I12)', iostat=ios) value
```

Comment:
You don't need to do this for rawstring.  You can just do a separate `HConfigAsString` for those.  Then you can ignore concerns about formatting.

Reply:
I did that originally, but then I tried to parse the string to produce the log message format we wanted:

    TY KEY VALUE[(RANK)
    
where TY is "CH", "L", "R4", "R8", "I4", or "I8" (The format can be tweaked as needed.) Since the string is coming from ESMF, I didn't see a way to control the format. So, I used the Fortran `write` to control the format of the value. The message is a comma-separated string for now until I have the logger part in place.